### PR TITLE
Experimental workaround for cdn03.quay.io flake

### PR DESCRIPTION
**Pull request by edsantiago**
I can't find instances of the "lookup cdn03.quay.io" flake
in RHEL gating tests. RHEL does not use systemd-resolved.
Maybe pure coincidence. Maybe not.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

File: contrib/cirrus/setup_environment.sh [hunk 1]
```diff
@@ -212,6 +212,33 @@ if [[ -n "$ROOTLESS_USER" ]]; then
...
+    if [[ -e $nsswitch ]]; then
+        if grep -q -E 'hosts:.*resolve' $nsswitch; then
+            msg "Disabling systemd-resolved"
+            sed -i -e 's/^\(hosts: *\).*/\1files dns myhostname/' $nsswitch
+            systemctl stop systemd-resolved
+            rm -f /etc/resolv.conf
+            systemctl start NetworkManager
+
+            # Sigh. The first time, NetworkManager doesn't seem
+            # to create a resolv.conf. Keep retrying until it does.
```
Comment (Luap99) [inline contrib/cirrus/setup_environment.sh:227]: You need to restart it, systemd will ignore start when a unit it is already running and AFAIK NetworkManager should already be running unless we explicitly disabled it use.

Comment (edsantiago) [reply contrib/cirrus/setup_environment.sh:227]: Hm, could be. On a 1minutetip  VM it was not started by default. Anyhow, no flakes in two runs, I'm going to keep hammering away at CI with this to see how it goes. Thanks for the suggestion.

**Comment by edsantiago** [comment]
Five (six?) runs, and no sign of the flake; but here's the flake, right now, in a different PR: https://cirrus-ci.com/task/5480576929497088

**Comment by edsantiago** [comment]
I lost count of the retries. None failed with the `cdn03` flake, despite other concurrent PRs  flaking that way.

@containers/podman-maintainers it makes me queasy to ask this, but please review and merge if you see fit. I'd like to reevaluate in a few weeks, see if the `cdn03` flake persists, and then figure out where to go from there.

**Review by vrothberg — APPROVED** [review]
LGTM

I added a reminder to my calendar.

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by vrothberg** [comment]
> LGTM
> 
> I added a reminder to my calendar.

Reminder kicked in.  @edsantiago, all good with this change?

**Comment by edsantiago** [comment]
Oh, interesting. Here's the `cdn03` flake list for the last 25 days:

* fedora-36-aarch64 : sys podman fedora-36-aarch64 root host
  * PR #17864
    * [03-21 05:49](https://api.cirrus-ci.com/v1/artifact/task/5165656102928384/html/sys-podman-fedora-36-aarch64-root-host.log.html)
* fedora-37-aarch64 : Build for fedora-37-aarch64
  * PR #17963
    * [03-28 09:56](https://api.cirrus-ci.com/v1/task/6132890992377856/logs/prebuild.log)
  * PR #17951
    * [03-27 17:21](https://api.cirrus-ci.com/v1/task/4739566255472640/logs/prebuild.log)
* fedora-37-aarch64 : machine podman fedora-37-aarch64 rootless host
  * PR #17875
    * [03-21 11:31](https://api.cirrus-ci.com/v1/task/6340341503098880/logs/main.log)
  * PR #17873
    * [03-21 10:26](https://api.cirrus-ci.com/v1/task/6750384111222784/logs/main.log)
  * PR #17810
    * [03-16 07:13](https://api.cirrus-ci.com/v1/task/5197685636988928/logs/main.log)
* fedora-37-aarch64 : sys podman fedora-37-aarch64 root host
  * PR #17925
    * [03-25 07:47](https://api.cirrus-ci.com/v1/artifact/task/4613156778016768/html/sys-podman-fedora-37-aarch64-root-host.log.html)
  * PR #17873
    * [03-21 13:57](https://api.cirrus-ci.com/v1/artifact/task/6543844066459648/html/sys-podman-fedora-37-aarch64-root-host.log.html)
    * [03-21 10:55](https://api.cirrus-ci.com/v1/artifact/task/5074728390492160/html/sys-podman-fedora-37-aarch64-root-host.log.html)
  * PR #17862
    * [03-21 06:20](https://api.cirrus-ci.com/v1/artifact/task/6001330909609984/html/sys-podman-fedora-37-aarch64-root-host.log.html)
    * [03-21 05:44](https://api.cirrus-ci.com/v1/artifact/task/6668424491827200/html/sys-podman-fedora-37-aarch64-root-host.log.html)
  * PR #17809
    * [03-16 07:41](https://api.cirrus-ci.com/v1/artifact/task/5265544593539072/html/sys-podman-fedora-37-aarch64-root-host.log.html)

...and of those, some are in "Build" step (which ISTR happens before my disable-systemd code?) and the rest are in `v4.4` branches. I would say this has been a success. Now what???

**Comment by vrothberg** [comment]
> Now what???

Can we disable it when building the VM images?

**Pull request merged**